Source/Core/Rectangle.js

```diff
@@ -1073,17 +1085,9 @@ define([
         }
 
         // Check if the rectangle crosses the IDL
-        if (!containsPole) {
-            unprojected.longitude = CesiumMath.PI;
-            unprojected.latitude = (result.north + result.south) * 0.5;
-            var projectedIdlCheckpoint = mapProjection.project(unprojected, projectedScratch);
-            var projectedIdlCartographic = projectedIdlScratch;
-            projectedIdlCartographic.longitude = projectedIdlCheckpoint.x;
-            projectedIdlCartographic.latitude = projectedIdlCheckpoint.y;
-            if (Rectangle.contains(projectedRectangle, projectedIdlCartographic)) {
-                result.west = idlCrossWest;
-                result.east = idlCrossEast;
-            }
+        if (!containsPole && crossedIdl) {
+            result.west = idlCrossWest;
+            result.east = idlCrossEast;
         }
 
         // Clamp
```

Comment:
I know it's not part of this PR, but why is this clamp here? Can't the unprojected rectangle be much larger than the projected rectangle? Perhaps, this should clamp to `Rectangle.MAX_VALUE`.

Reply:
There's a lot of projections where it's really easy to project to infinity, so we wanted to clamp to either the projections boundaries or `Rectangle.MAX_VALUE` (see above).